cpluginv1/internal/fromproto/specifier.go

```diff
@@ -65,6 +82,19 @@ func SpecifierParameter(in *connectorv1.Specifier_Parameter) (cpluginv1.Specifie
 		Default:     in.Default,
 		Required:    in.Required,
 		Description: in.Description,
+		Type:        cpluginv1.ParameterType(in.Type),
+		Validations: SpecifierParameterValidations(in.Validations),
 	}
 	return out, nil
 }
+
+func SpecifierParameterValidations(in []*connectorv1.Specifier_Parameter_Validation) []cpluginv1.ParameterValidation {
+	out := make([]cpluginv1.ParameterValidation, 1)
+	for _, v := range in {
+		out = append(out, cpluginv1.ParameterValidation{
+			Type:  cpluginv1.ValidationType(v.Type),
+			Value: v.Value,
+		})
```

Comment:
Watch out when initializing the slice - you're creating a slice that already contains 1 element and then appending to it. The first element is always going to be an empty parameter, we don't want that. The best thing would be to create a slice of length `len(in)` and then set elements directly `out[i] = ...`, this way we avoid unnecessary resizes of the backing array that `append` does behind the scenes.

---

That said! I have no idea why I added functions for converting responses from proto to internal types, this code is only used by the server side (plugin side). Feels like I wasn't sure at that point if we'd need it or not. You can also clean up the code and just remove anything related to responses in `fromproto` and anything related to requests in `toproto`, I don't think we will ever need that code.